### PR TITLE
refactor: Define PaymentChannelClaim transaction model

### PR DESCRIPTION
**Pull request by natenichols**
## High Level Overview of Change
Add a transaction model for `PaymentChannelClaim`

### Context of Change
Adds a typescript type for `PaymentChannelClaim` interface, a `PaymentChannelClaimFlags `interface, a `verifyPaymentChannelClaim()` function for runtime verification, and tests for `verifyPaymentChannelClaims`.

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

File: src/models/transactions/paymentChannelClaim.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export interface PaymentChannelClaim extends BaseTransaction {
+    TransactionType: "PaymentChannelClaim";
+    Flags?: number | PaymentChannelClaimFlags
+    Channel: string;
+    Balance?: string;
+    Amount?: string;
+    Signature?: string;
+    PublicKey?: string;
```
Comment (mukulljangid) [inline src/models/transactions/paymentChannelClaim.ts:16]: nit: remove semicolons

File: src/models/transactions/paymentChannelClaim.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { BaseTransaction, GlobalFlags, verifyBaseTransaction } from "./common";
+
+export interface PaymentChannelClaimFlags extends GlobalFlags {
+    tfRenew?: boolean;
+    tfClose?: boolean;
+}
+
+export interface PaymentChannelClaim extends BaseTransaction {
+    TransactionType: "PaymentChannelClaim";
+    Flags?: number | PaymentChannelClaimFlags
```
Comment (JST5000) [inline src/models/transactions/paymentChannelClaim.ts:11]: Does this overwrite the `| GlobalFlags` type option in BaseTransaction?

**Review by khancode — CHANGES_REQUESTED** [review]
Suggesting the same changes to be made just like with [EscrowCancel PR](https://github.com/ripple/ripple-lib/pull/1511#pullrequestreview-733379253):
- make tests easier to write/read (using beforeEach())
- make error messages more concise
- nit: remove semicolons

**Review by JST5000 — APPROVED** [review]
LGTM (Just had one point of confusion, but the number type should cover it anyway)

**Review by khancode — APPROVED** [review]
LGTM

File: src/models/transactions/paymentChannelClaim.ts (hunk 2 above)
Comment (mvadari) [reply src/models/transactions/paymentChannelClaim.ts:11]: Wondering the same thing - is this necessary given the `extends GlobalFlags`?

**Review by mvadari — APPROVED** [review]
Make sure you re-rebase on 2.0

File: src/models/transactions/paymentChannelClaim.ts (hunk 2 above)
Comment (natenichols) [reply src/models/transactions/paymentChannelClaim.ts:11]: Yeah, this should enforce that a PaymentChannelClaim has a PaymentChannelClaimFlags

**Pull request merged**